src/CashuWallet.ts

```diff
@@ -474,22 +473,43 @@ class CashuWallet {
 		}
 		const sendBlindedMessages = this.createRandomBlindedMessages(amount, preference, counter);
 
-		// join keepBlindedMessages and sendBlindedMessages
+		const convertedKeepMessages = keepBlindedMessages.blindedMessages.map((_, i) => ({
+			msg: keepBlindedMessages.blindedMessages[i],
+			secret: keepBlindedMessages.secrets[i],
+			rs: keepBlindedMessages.rs[i],
+			amount: keepBlindedMessages.amounts[i],
+			keep: true
+		}));
+
+		const convertedSendMessages = sendBlindedMessages.blindedMessages.map((_, i) => ({
+			msg: sendBlindedMessages.blindedMessages[i],
+			secret: sendBlindedMessages.secrets[i],
+			rs: sendBlindedMessages.rs[i],
+			amount: sendBlindedMessages.amounts[i],
+			keep: false
+		}));
+		const keepVector: Array<boolean> = [];
 		const blindedMessages: BlindedTransaction = {
-			blindedMessages: [
-				...keepBlindedMessages.blindedMessages,
-				...sendBlindedMessages.blindedMessages
-			],
-			secrets: [...keepBlindedMessages.secrets, ...sendBlindedMessages.secrets],
-			rs: [...keepBlindedMessages.rs, ...sendBlindedMessages.rs],
-			amounts: [...keepBlindedMessages.amounts, ...sendBlindedMessages.amounts]
+			blindedMessages: [],
+			secrets: [],
+			rs: [],
+			amounts: []
```

Comment:
we could add the keep vector here instead

Reply:
True, I was not sure how many lines of code depend on the type  `BlindedTransaction` and wanted to reduce this PRs footprint. But I will check that out in my next iteration.